static/app/components/organizations/multipleEnvironmentSelector.tsx

```diff
@@ -54,204 +49,154 @@ type Props = WithRouterProps & {
   customLoadingIndicator?: React.ReactNode;
   detached?: boolean;
   forceEnvironment?: string;
-} & DefaultProps;
-
-type State = {
-  hasChanges: boolean;
-  selectedEnvs: Set<string>;
 };
 
 /**
  * Environment Selector
  *
  * Note we only fetch environments when this component is mounted
  */
-class MultipleEnvironmentSelector extends React.PureComponent<Props, State> {
-  static defaultProps: DefaultProps = {
-    value: [],
-  };
-
-  state: State = {
-    selectedEnvs: new Set(this.props.value),
-    hasChanges: false,
-  };
-
-  componentDidUpdate(prevProps: Props) {
-    // Need to sync state
-    if (this.props.value !== prevProps.value) {
-      this.syncSelectedStateFromProps();
-    }
-  }
-
-  syncSelectedStateFromProps = () =>
-    this.setState({selectedEnvs: new Set(this.props.value)});
+function MultipleEnvironmentSelector({
+  loadingProjects,
+  onChange,
+  onUpdate,
+  organization,
+  projects,
+  selectedProjects,
+  value,
+  customDropdownButton,
+  customLoadingIndicator,
+  detached,
+  forceEnvironment,
+  router,
+}: Props) {
+  const [selectedEnvs, setSelectedEnvs] = useState(value);
+
+  // Update selected envs value on change
+  useEffect(() => setSelectedEnvs(value), [value]);
+
+  // We keep a separate list of selected environments to use for sorting. This
+  // allows us to only update it after the list is closed, to avoid the list
+  // jumping around while selecting projects.
```

Comment:
Is it worth noting that this requires the MultipleEnvironmentSelector to be removed and readded to the DOM?

Reply:
It doesn't, the `lastSelectedEnvs` ref is updated when we close it.

Unless I'm misunderstanding your comment?